packages/lib/isOutOfBounds.tsx

```diff
@@ -219,19 +240,30 @@ export function isTimeViolatingFutureLimit({
   periodLimits: PeriodLimits;
 }) {
   const log = logger.getSubLogger({ prefix: ["isTimeViolatingFutureLimit"] });
-  const date = dayjs(time);
-  if (periodLimits.rollingEndDay) {
-    const isAfterRollingEndDay = date.isAfter(periodLimits.rollingEndDay);
-    log.debug({
-      formattedDate: date.format(),
+  // Because we are just going to compare times, we need not convert the time to a particular timezone
+  // date.isAfter/isBefore are timezone neutral
+  const dateInSystemTz = dayjs(time);
+  if (periodLimits.endOfRollingPeriodEndDayInBookerTz) {
+    const isAfterRollingEndDay = dateInSystemTz.isAfter(periodLimits.endOfRollingPeriodEndDayInBookerTz);
+    log.debug("rollingEndDayCheck", {
+      formattedDate: dateInSystemTz.format(),
       isAfterRollingEndDay,
-      rollingEndDay: periodLimits.rollingEndDay.format(),
+      endOfRollingPeriodEndDayInBookerTz: periodLimits.endOfRollingPeriodEndDayInBookerTz.format(),
     });
     return isAfterRollingEndDay;
   }
 
-  if (periodLimits.rangeStartDay && periodLimits.rangeEndDay) {
-    return date.isBefore(periodLimits.rangeStartDay) || date.isAfter(periodLimits.rangeEndDay);
+  if (periodLimits.startOfRangeStartDayInEventTz && periodLimits.endOfRangeEndDayInEventTz) {
+    const isBeforeRangeStart = dateInSystemTz.isBefore(periodLimits.startOfRangeStartDayInEventTz);
```

Comment:
As explained in the comment in the code, we can compare system tz date object with event timezone date object.